### PR TITLE
Added SAP monitor resource type to HanaOnAzure RP

### PR DESCRIPTION
**Pull request by PakDLiu**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by amarzavery** [comment]
LGTM. Will wait for ARM feedback, since new apis have been added

**Review by amarzavery — APPROVED** [review]

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2017-11-03-preview/hanaonazure.json [hunk 1]
```diff
@@ -426,6 +426,103 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HanaOnAzure/sapMonitors/{sapMonitorName}": {
```
Comment (majastrz) [inline specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2017-11-03-preview/hanaonazure.json]: >/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HanaOnAzure/sapMonitors/ [](start = 5, length = 111)

As per the RPC, you also need a route that returns a list of all sapMonitors. ([https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#get-resource](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#get-resource))

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2017-11-03-preview/hanaonazure.json [hunk 2]
```diff
@@ -426,6 +426,103 @@
...
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
```
Comment (majastrz) [inline specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2017-11-03-preview/hanaonazure.json]: >put [](start = 7, length = 3)

Tracked resources also require implementing patch. Patch should at minimum allow you to update tags on the resource.

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2017-11-03-preview/hanaonazure.json [hunk 3]
```diff
@@ -821,6 +918,69 @@
...
+          "description": "Database port of the HANA instance."
+        },
+        "hanaDbUsername": {
+          "type": "string",
+          "description": "Database username of the HANA instance."
+        },
+        "hanaDbPassword": {
+          "type": "string",
+          "description": "Database password of the HANA instance."
+        },
```
Comment (majastrz) [inline specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2017-11-03-preview/hanaonazure.json]: Will this ever be returned on GET responses?

**Review by majastrz — COMMENTED** [review]
I added comments. Please take a look. The current on-call will continue the review.

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2017-11-03-preview/hanaonazure.json (hunk 3 above)
Comment (PakDLiu) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2017-11-03-preview/hanaonazure.json]: No, it won't.
On the GET, will not exist.

Is there a way to reuse the definition with slight changes?

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2017-11-03-preview/hanaonazure.json (hunk 2 above)
Comment (PakDLiu) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2017-11-03-preview/hanaonazure.json]: done, added patch

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2017-11-03-preview/hanaonazure.json (hunk 1 above)
Comment (PakDLiu) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2017-11-03-preview/hanaonazure.json]: done, added list

**Comment by PakDLiu** [comment]
There seems to be a spell check test that is failing on a file I didn't edit...

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Comment by ravbhatnagar** [comment]
Signing off from ARM side. Comments from previous review have been addressed.

**Review by amarzavery — APPROVED** [review]

**Pull request merged**